src/Uno.UI/UI/Xaml/Media/Imaging/RenderTargetBitmap.cs

```diff
@@ -14,14 +14,89 @@
 using Uno.UI.Xaml.Media;
 using Buffer = Windows.Storage.Streams.Buffer;
 using System.Buffers;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.UI.Xaml.Media.Imaging
 {
 #if NOT_IMPLEMENTED
 	[global::Uno.NotImplemented("IS_UNIT_TESTS", "__WASM__", "__NETSTD_REFERENCE__")]
 #endif
-	public partial class RenderTargetBitmap : ImageSource, IDisposable
+	public partial class RenderTargetBitmap : ImageSource
 	{
+#if !__ANDROID__
+		// This is to avoid LOH array allocations
+		private unsafe struct UnmanagedArrayOfBytes : IDisposable
+		{
+			public nint Pointer;
+			public int Length { get; }
+
+			public UnmanagedArrayOfBytes(int length)
+			{
+				Length = length;
+				Pointer = Marshal.AllocHGlobal(length);
+			}
+
+			public byte this[int index]
+			{
+				get
+				{
+					return ((byte*)Pointer.ToPointer())[index];
+				}
+				set
+				{
+					((byte*)Pointer.ToPointer())[index] = value;
+				}
+			}
+
+			public void Dispose()
+			{
+				Marshal.FreeHGlobal(Pointer);
+			}
+		}
+
+		// https://stackoverflow.com/questions/52190423/c-sharp-access-unmanaged-array-using-memoryt-or-arraysegmentt
+		private sealed unsafe class UnmanagedMemoryManager<T> : MemoryManager<T>
+			where T : unmanaged
+		{
+			private readonly T* _pointer;
+			private readonly int _length;
+
+			/// <summary>
+			/// Create a new UnmanagedMemoryManager instance at the given pointer and size
+			/// </summary>
+			public UnmanagedMemoryManager(T* pointer, int length)
+			{
+				if (length < 0) throw new ArgumentOutOfRangeException(nameof(length));
+				_pointer = pointer;
+				_length = length;
+			}
+			/// <summary>
+			/// Obtains a span that represents the region
+			/// </summary>
+			public override Span<T> GetSpan() => new Span<T>(_pointer, _length);
+
+			/// <summary>
+			/// Provides access to a pointer that represents the data (note: no actual pin occurs)
+			/// </summary>
+			public override MemoryHandle Pin(int elementIndex = 0)
+			{
+				if (elementIndex < 0 || elementIndex >= _length)
+					throw new ArgumentOutOfRangeException(nameof(elementIndex));
+				return new MemoryHandle(_pointer + elementIndex);
```

Comment:
Is the returned MemoryHandle disposed properly?

Reply:
@jeromelaban It's probably caller's responsibility, e.g:

https://github.com/unoplatform/uno/blob/c500ee9d04f9e2d4fc9e959674adb246afd11fab/src/Uno.UI.Composition/Composition/SkiaCompositionSurface.skia.cs#L69

I'll try to do refactoring that will let us do the work without `MemoryManager<T>` at all